homeassistant/components/legrandinone.py

```diff
@@ -0,0 +1,480 @@
+        else:
+            device_type = event.get('who')
+
+            # If device is not yet known, register with platform (if loaded)
+            if event_type in hass.data[DATA_DEVICE_REGISTER]:
+                if device_type in hass.data[DATA_DEVICE_REGISTER][event_type]:
+                    _LOGGER.debug('device_id not known, adding new device')
+
+                    hass.data[DATA_ENTITY_LOOKUP][event_type][
+                        event_id].append(TMP_ENTITY.format(event_id))
+                    hass.async_create_task(
+                        hass.data[DATA_DEVICE_REGISTER][event_type][
+                            device_type](event))
+
+                else:
+                    _LOGGER.debug(
+                        'device_id not known and automatic %s add disabled',
+                        device_type)
+            else:
+                _LOGGER.debug('device_id not known and automatic add disabled')
+
+    # When connecting to tcp host instead of serial port (optional)
+    host = config[DOMAIN].get(CONF_HOST)
+    # TCP port when host configured, otherwise serial port
+    port = config[DOMAIN][CONF_PORT]
+
+    @callback
+    def reconnect(exc=None):
+        """Schedule reconnect after connection has been unexpectedly lost."""
+        # Reset protocol binding before starting reconnect
+        hass.data[IOBL_PROTOCOL_HANDLE] = None
+
+        async_dispatcher_send(hass, SIGNAL_AVAILABILITY, False)
+
+        # If HA is not stopping, initiate new connection
+        if hass.state != CoreState.stopping:
+            _LOGGER.warning('disconnected from Iobl, reconnecting')
+            hass.async_create_task(connect())
+
+    async def connect():
+        """Set up connection and hook it into HA for reconnect/shutdown."""
+        _LOGGER.info('Initiating Iobl connection')
+
+        # IOBL create_iobl_connection decides based on the value of host
+        # (string or None) if serial or tcp mode should be used
+
+        # Initiate serial/tcp connection to IOBL gateway
+        connection = create_iobl_connection(
+            port=port,
+            host=host,
+            event_callback=event_callback,
+            disconnect_callback=reconnect,
+            loop=hass.loop,
+            ignore=None
+        )
+
+        try:
+            with async_timeout.timeout(CONNECTION_TIMEOUT,
+                                       loop=hass.loop):
+                transport, protocol = await connection
+
+        except (serial.serialutil.SerialException, ConnectionRefusedError,
+                TimeoutError, OSError, asyncio.TimeoutError) as exc:
+            reconnect_interval = config[DOMAIN][CONF_RECONNECT_INTERVAL]
+            _LOGGER.exception(
+                "Error connecting to iobl, reconnecting in %s",
+                reconnect_interval)
+            # Connection to IOBL device is lost, make entities unavailable
+            async_dispatcher_send(hass, SIGNAL_AVAILABILITY, False)
```

Comment:
It's enough to send one signal for no availability. It's done in the `reconnect` callback also. We can remove it here.

Reply:
Done.